src/Raven.Server/Documents/Indexes/Persistence/Lucene/OutputReduceIndexWriteOperation.cs

```diff
@@ -30,10 +30,10 @@ public OutputReduceIndexWriteOperation(MapReduceIndex index, LuceneVoronDirector
 
         public override void Commit(IndexingStatsScope stats)
         {
-            var enqueue = DocumentDatabase.TxMerger.Enqueue(_outputReduceToCollectionCommand);
-            base.Commit(stats);
             try
             {
+                var enqueue = DocumentDatabase.TxMerger.Enqueue(_outputReduceToCollectionCommand);
+                base.Commit(stats);
```

Comment:
OK. 
So I will throw in the catch after adding the alert. 
This way we will retry the batch if we had an error, either commit to the docs store or to the lucene store, instead of continue to index without creating some output documents.
@ayende please also take a look.